apps/ai-content-generator/src/components/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, CopyButton, Tabs } from '@contentful/f36-components';
+import useAI from '@hooks/dialog/useAI';
+import TextFieldWithButtons from '@components/common/text-field-with-buttons/TextFieldWIthButtons';
+import { OutputTab } from '../../Output';
+
+interface Props {
+  generate: () => void;
+  ai: ReturnType<typeof useAI>;
+  hasOutputField: boolean;
+  apply: () => void;
+}
+
+const GeneratedTextPanel = (props: Props) => {
+  const { generate, ai, hasOutputField, apply } = props;
+  const { sendStopSignal, output, setOutput, isGenerating } = ai;
+
+  const handleGeneratedTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOutput(event.target.value);
+  };
+
+  return (
+    <Tabs.Panel id={OutputTab.GENERATED_TEXT}>
+      {isGenerating && (
+        <TextFieldWithButtons inputText={output}>
+          <Button onClick={sendStopSignal}>Stop Generating</Button>
+        </TextFieldWithButtons>
+      )}
+
+      {!isGenerating && (
+        <TextFieldWithButtons inputText={output} onFieldChange={handleGeneratedTextChange}>
+          <>
+            <CopyButton value={output} />
+            <Button onClick={generate}>Regenerate</Button>
+            <Button isDisabled={hasOutputField} onClick={apply}>
+              Apply
+            </Button>
+          </>
+        </TextFieldWithButtons>
+      )}
```

Comment:
nit - could either way here around if a ternary is more readable.` isGenerating ? ___ : ___`

Reply:
I can go either way on this too. Let's use the ternary!